File: Sorting_Algorithms/Python/bubble_sort.py
# -*- coding: utf-8 -*-
"""
Created on Fri Jul 10 17:08:23 2020

@author: Soham Shah
"""

# Bubble Sort - Iterative

arr = [64, 34, 25, 12, 22, 11, 90] 

for i in range(len(arr)-1):
    for j in range(i+1, len(arr)):
        if arr[i] > arr[j]:
            arr[i], arr[j] = arr[j], arr[i]
            
# Bubble Sort - Optimized

for i in range(len(arr)-1):
    swapped = False
    for j in range(i+1, len(arr)):
        if arr[i] > arr[j]:
            arr[i], arr[j] = arr[j], arr[i]
            swapped = True
    if swapped == False:
        break

# Bubble Sort Recursive
        
def bubble_sort(listt): 
    for i, num in enumerate(listt): 
        try: 
            if listt[i+1] < num: 
                listt[i] = listt[i+1] 
                listt[i+1] = num 
                bubble_sort(listt) 
        except IndexError: 
            pass
    return listt 
  
bubble_sort(arr)

  
        
            
            

    

File: Sorting_Algorithms/C++/bubble_sort.cpp
// Bubble Sort
#include <bits/stdc++.h>
using namespace std;

void Array(int arr[], int n){
	int i;
	for(i=0; i < n; i++){
		cout << arr[i] << " ";
		
	}
}

void swap(int *a, int *b){
	int temp = *a;
	*a = *b;
	*b = temp;
}

void BubbleSort(int arr[],int n){
	int i,j, swapped;
	for(i=0;i<n-1;i++){
	
		for(j=0; j<n-i-1; j++){
			if(arr[j] > arr[j+1]){
				swap(&arr[j], &arr[j+1]);
				swapped=1;
			}
			
		}
	}
}
int main(){
	int arr[1000];
	int i, ele;
	cout << "Enter number of elements: \n";
	cin >> ele;
	cout << "Enter Elements of the Array!\n";
	for(i=0; i<ele; i++){
		cin >> arr[i];
	}
	BubbleSort(arr, ele);
	cout << "Sorted Array is: \n";
	Array(arr, ele);
	return 0;		
}


